### PR TITLE
feat(expext): toBeOK for APIResponse

### PR DESCRIPTION
**Pull request by yury-s**
#10189

### CODE REVIEW

File: packages/playwright-test/src/matchers/matchers.ts [hunk 1]
```diff
@@ -263,3 +268,15 @@ export function toHaveURL(
...
   }, expected, options);
 }
+
+export async function toBeOK(
+  this: ReturnType<Expect['getState']>,
+  response: APIResponseEx
+) {
+  const matcherName = 'toBeOK';
+  expectType(response, 'APIResponse', matcherName);
+  const log = await response._fetchLog();
```
Comment (dgozman) [inline packages/playwright-test/src/matchers/matchers.ts:278]: I think we should skip fetching the log when `this.isNot !== response.ok()`.

File: packages/playwright-test/types/testExpect.d.ts [hunk 2]
```diff
@@ -103,6 +103,11 @@ declare global {
        */
       toBeHidden(options?: { timeout?: number }): Promise<R>;
 
+      /**
+       * Asserts given APIResponse's status is 200.
```
Comment (dgozman) [inline packages/playwright-test/types/testExpect.d.ts:107]: between 200 and 299

**Review by dgozman — APPROVED** [review]

File: tests/playwright-test/playwright.expect.true.spec.ts [hunk 3]
```diff
@@ -322,3 +322,41 @@ test('should print syntax error', async ({ runInlineTest }) => {
...
 });
+
+test('should support toBeOK', async ({ runInlineTest, server }) => {
+  const result = await runInlineTest({
+    'a.test.ts': `
+      const { test } = pwt;
+
+      test('pass with response', async ({ page }) => {
+        const res = await page.request.get('${server.EMPTY_PAGE}');
+        expect(res).toBeOK();
```
Comment (mxschmitt) [inline tests/playwright-test/playwright.expect.true.spec.ts:333]: You need await there afaik. Otherwise it ends up in an unhanded promise rejection.

**Review by mxschmitt — APPROVED** [review]
Should we add it to test-assertions-js.md and create a new class-responseassertions.md for it?a

File: tests/playwright-test/playwright.expect.true.spec.ts (hunk 3 above)
Comment (yury-s) [reply tests/playwright-test/playwright.expect.true.spec.ts:333]: Yeah, good point.

**Comment by yury-s** [comment]
Yes, I'll add it to the docs.

**Pull request merged**